packages/sd-jwt-vc/src/sd-jwt-vc-instance.ts

```diff
@@ -113,12 +123,167 @@ export class SDJwtVcInstance extends SDJwtInstance<SdJwtVcPayload> {
         };
       });
 
+    await this.verifyStatus(result);
+    if (this.userConfig.loadTypeMetadataFormat) {
+      await this.verifyVct(result);
+    }
+    return result;
+  }
+
+  /**
+   * Default function to fetch the VCT from the uri. We assume that the vct is a URL that is used to fetch the VCT.
+   * @param uri
+   * @returns
+   */
+  private async vctFetcher(
+    uri: string,
+    integrity?: string,
+  ): Promise<TypeMetadataFormat> {
+    // modify the uri based on https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-04.html#section-6.3.1
+    const elements = uri.split('/');
+    //insert a new element on the thrid position, but not replace it
+    elements.splice(3, 0, '.well-known/vct');
+    const url = elements.join('/');
+    return this.fetch<TypeMetadataFormat>(url, integrity);
+  }
+
+  /**
+   * Validates the integrity of the response if the integrity is passed. If the integrity does not match, an error is thrown.
+   * @param integrity
+   * @param response
+   */
+  private async validateIntegrity(
+    response: Response,
+    url: string,
+    integrity?: string,
+  ) {
+    if (integrity) {
+      // validate the integrity of the response according to https://www.w3.org/TR/SRI/
+      const arrayBuffer = await response.arrayBuffer();
+      const alg = integrity.split('-')[0];
+      //TODO: error handling when a hasher is passed that is not supporting the required algorithm acording to the spec
+      const hashBuffer = await (this.userConfig.hasher as Hasher)(
+        arrayBuffer,
+        alg,
+      );
+      const integrityHash = integrity.split('-')[1];
+      const hash = Array.from(new Uint8Array(hashBuffer))
+        .map((byte) => byte.toString(16).padStart(2, '0'))
+        .join('');
+      if (hash !== integrityHash) {
+        throw new Error(
+          `Integrity check for ${url} failed: is ${hash}, but expected ${integrityHash}`,
+        );
+      }
+    }
+  }
+
+  /**
+   * Fetches the content from the url with a timeout of 10 seconds.
+   * @param url
+   * @returns
+   */
+  private async fetch<T>(url: string, integrity?: string) {
+    const controller = new AbortController();
+    const timeoutId = setTimeout(() => controller.abort(), 10000);
+    try {
+      const response = await fetch(url, {
```

Comment:
Oh RN also support the fetch function. Good :)

Reply:
Since fetch is supported in the browser and in node, we can use it for both. I see no point that we need it to override with a custom implementation (at this point)

Reply:
Yeah I agree.

Reply:
fetch is also used in the statuslist fetcher, but the whole method can be overwritten via userconfig. So I think it's fine for now. Otherwise we need to implement a general fetch function that is covering all cases.